SettingsSectionAccount

franarave opened the pull request:
https://github.com/DuckyTeam/ducky-web/issues/1585

franarave committed:
SettingsSectionAccount

https://github.com/DuckyTeam/ducky-web/issues/1585

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+        onClick={props.onClick}
+        onCloseClick={props.onCloseClick}
+        onFocus={props.onFocus}
+        onRadioButtonClicked={props.onRadioButtonClicked}
```
onRadioButtonClicked is an error inside SettingsItemPassword. You can remove from here and the PropType validation on this component and SettingsItemPassword.

franarave replied:
done 👍 removed from both components.

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+        />
+      <SettingsItemEmail
+        email={props.email}
+        emailOnClick={props.emailOnClick}
```
`emailOnClick={props.emailOnClick}` wont work. On the left of the equals we need the name of the prop in the parent component and on the right we can give the prop a new name for this component. 
So in this case we say `onClick={props.emailOnClick}` since the parent component is expecting to receive something on `onClick`

franarave replied:
ok!

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+        oldPassword={props.oldPassword}
+        onBlur={props.onBlur}
+        onChange={props.onChange}
+        onClick={props.onClick}
```
similar to above at `onClick={props.emailOnClick}`lets rename the onClick prop for this component to be something a bit more explicit and understandable about what is to be clicked.

franarave replied:
done 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+        textAreaLength={props.textAreaLength}
+        textSize={props.textSize}
+        />
+      <Wrapper className={classNames(styles.wrapper,
```
it is a bit odd to have `classNames` at a child inside a component. So maybe it is best suited to using it in the top level `<div>`

franarave replied:
done 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+        emailOnClick={props.emailOnClick}
+        />
+      <SettingsItemPassword
+        className={classNames({[props.className]: props.className})}
```
there is no need for this line unless you want to pass in new styling.

franarave replied:
you're right... sorry 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+            {moment(props.date).format("MMMM YYYY")}
+        </Typography>
+      </Wrapper>
+
```
remove blank lines

franarave replied:
done 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+}
+
+SettingsSectionAccount.propTypes = {
+  children: PropTypes.string,
```
`children` prop is not used

franarave replied:
removed 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+  oldPassword: PropTypes.string,
+  onBlur: PropTypes.func,
+  onChange: PropTypes.func,
+  onClick: PropTypes.func,
```
as above rename to be more explicit

franarave replied:
done 👍

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+  onClick: PropTypes.func,
+  onCloseClick: PropTypes.func,
+  onFocus: PropTypes.func,
+  onRadioButtonClicked: PropTypes.func,
```
`onRadioButtonClicked` prop is not needed

franarave replied:
Removed

nathan-bird commented on src/SettingsSectionAccount/index.js:
```diff
@@ -0,0 +1,87 @@
+  textSize: PropTypes.string,
+  theme: PropTypes.string,
+  title: PropTypes.string,
+  type: PropTypes.string
```
`type` prop is not used

franarave replied:
removed

nathan-bird commented on src/SettingsSectionAccount/stories.js:
```diff
@@ -0,0 +1,28 @@
+  textAreaLength: number(50),
+  textSize: oneOf('default', 'large', 'huge'),
+  text: string('Aktiv siden '),
+  date: number(1319191481000)
```
grab your updated props from the component

nathan-bird commented on src/SettingsSectionAccount/styles.css:
```diff
@@ -0,0 +1,35 @@
+background-color: #f4f7f9;
+
+}
+
```
simple code cleanup could improve readability. eg consistent tabs, spaces after css property, etc

franarave replied:
done 👍

nathan-bird reviewed (changes requested):
Looks good. Some simple prop stuff to sort out.
Ask me if you need any more info :)
Keep it up!!

franarave committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into SettingsActionAccount

franarave committed:
Settings Section Account

https://github.com/DuckyTeam/ducky-web/issues/1585
Fix Settings Item Password (prop unused removed)

nathan-bird committed:
onClick fix
